syntax = "proto3";

import "google/protobuf/timestamp.proto";

option java_multiple_files = true;
option java_package = "io.grpc.examples.pipelinescheduler";
option java_outer_classname = "PipelineSchedulerProto";
option objc_class_prefix = "HLW";

package pipelinescheduler;

// The service definition.
service DataTransferService {
  rpc GpuPointerTransfer (GpuPointerPayload) returns (SimpleConfirm) {}
  rpc SharedMemTransfer (SharedMemPayload) returns (SimpleConfirm) {}
  rpc SerializedDataTransfer (SerializedDataPayload) returns (SimpleConfirm) {}
}

// The payload messages
message GpuPointerPayload {
  int64 timestamp = 1;
  int32 width = 2;
  int32 height = 3;
  uint32 slo = 4;
  string path = 5;

  bytes pointer = 99;
}

message SharedMemPayload {
  int64 timestamp = 1;
  int32 width = 2;
  int32 height = 3;
  uint32 slo = 4;
  string path = 5;

  string name = 99;
}

message SerializedDataPayload {
  int64 timestamp = 1;
  uint32 datalen = 2;
  int32 width = 3;
  int32 height = 4;
  uint32 slo = 5;
  string path = 6;

  bytes data = 99;
}


// The response message confirming transfer
message SimpleConfirm {
}
